sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -480,7 +480,7 @@ public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellatio
                 activeProducer = PartitionProducers.GetOrAdd(eventBatch.SendOptions.PartitionId, id => Connection.CreateTransportProducer(id, RetryPolicy));
             }
 
-            using DiagnosticScope scope = CreateDiagnosticScope();
+            using DiagnosticScope scope = CreateDiagnosticScope(eventBatch.AsEnumerable<EventData>());
```

Comment:
I think we should consider holding onto the diagnostic identifiers in the `EventDataBatch` as they are `TryAdd`ed in.   The translation to enumerable isn't terribly expensive, but at some point, I will get around to fixing the batch such that it doesn't hold onto the raw `EventData` but rather the underlying transport message (`AmpqMessage` in this case)

Something like:

```csharp
public class EventDataBatch : IDisposable
{
    private List<string> DiagnosticIdentifiers { get; } = new List<string>();
    // ...
    
    internal IReadOnlyList<string> EventDiagnosticIdentifiers => DiagnosticIdentifiers;

    public bool TryAdd(EventData eventData)
    {
        bool instrumented = EventDataInstrumentation.InstrumentEvent(eventData, FullyQualifiedNamespace, EventHubName);
        bool added = InnerBatch.TryAdd(eventData);

        if (!added && instrumented)
        {
            EventDataInstrumentation.ResetEvent(eventData);
        }
        else if (EventDataInstrumentation.TryExtractDiagnosticId(eventData, out string diagnosticId))
        {
              DiagnosticIdentifiers.Add(diagnosticId);
        }

        return added;
    }
}
```